### ISSUE TITLE
Download Service Issue

### ISSUE THREAD

**Issue by Raghav-intrigue** [kiwix/kiwix-android#658]
### Environment
  * version of the software you use : 2.2
  * device / operating system : 6.0.1 Onplus X
  
### The Bug:

  * Content download does not start/or happens at a very slow speed if you switch apps just after starting the download. This was tested for different types of content and multiple times. The speed of download is significantly slower if the app gets minimized and the download needs to be canceled and restarted again for faster download.

* Content stays in the downloading tab with 100% progress even after completion of the download. I haven't been able to reliably reproduce this issue though.

  * This occurs if you don't wait for the progress bar on the download to move (which is around 8-12 seconds but can vary) after putting your content for downloading before switching apps or minimizing it.

*The first and second picture shows that the Wikipedia Math has been fully downloaded and yet still stays in the downloading tab. 
![pic 1](https://user-images.githubusercontent.com/12858001/37804943-0f23b41c-2e5d-11e8-90ea-ac10d55040cf.jpeg)
![pic 2](https://user-images.githubusercontent.com/12858001/37804945-124f0c90-2e5d-11e8-82ce-46e118f7f409.jpeg)

*These show the main issue of content not downloading if the app is swapped just after download starts. The download size of both was comparable, yet the one started afterwards downloaded first.

![p3](https://user-images.githubusercontent.com/12858001/37834445-bc685314-2ed3-11e8-9ab5-98114757e3ed.jpeg)
![p4](https://user-images.githubusercontent.com/12858001/37834446-bcb29d2a-2ed3-11e8-8c65-1bc1f23b45ac.jpeg)
![p5](https://user-images.githubusercontent.com/12858001/37834447-bcf671f8-2ed3-11e8-8a5a-131aa728e096.jpeg)

#### Steps to reproduce:

  1. Put a content (which you haven't downloaded before) on download
  2. Within seconds press the home button on your mobile to exit the app.
  3. The download will stall or will appear to happen at a very slow speed.
  4. Only canceling and redownloading will fix this, pausing and restarting does not work.

*Optimal behavior: 
The service should be able to handle the download in the background once it has started.

**Comment by arpank10** [comment]
This does not happen in my phone.

**Comment by Raghav-intrigue** [comment]
It can be a bit unreliable to reproduce but I tried it on the emulator and it still happened.  
I have attached the screenshots. Just for context, I have a gigabit connection and the file size is ~110 Mb.

![bug1](https://user-images.githubusercontent.com/12858001/37852271-fded3776-2f07-11e8-9eb2-950cfd876d7c.png)
The progress after 4 minutes:
![bug2](https://user-images.githubusercontent.com/12858001/37852274-fe1e86aa-2f07-11e8-8134-e24da28770ee.png)

But when canceled and restarted normally the same file downloaded within 30 seconds:
![normal1](https://user-images.githubusercontent.com/12858001/37852341-37213f60-2f08-11e8-8218-e581d32997ae.png)
![normal2](https://user-images.githubusercontent.com/12858001/37852342-375239d0-2f08-11e8-8fa5-af47721ab58b.png)
